back/app/models/user.rb

```diff
@@ -545,7 +545,10 @@ def validate_not_duplicate_email
   def validate_can_update_email
     return unless persisted? && (new_email_changed? || email_changed?)
 
-    if no_password? && confirmation_required?
+    # no_password? - here it's only for light registration
+    # confirmation_required? - it's always false for SSO providers that return email (all except ClaveUnica)
+    # !sso? - exclude ClaveUnica registrations (no email)
+    if no_password? && confirmation_required? && !sso?
```

Comment:
With light registration there is a situation where previously registered sso users can choose to enter their email address and get a verification code to perform an action, therefore requiring confirmation and opening up that security hole. It's a bit of an edge case though, so happy for that to be tackled in a later ticket.